### PR TITLE
Add Comment entity

### PR DESCRIPTION
**Pull request by HinataIzayoi**
Add /v1/poems/comment to allow User to add Comment to Poem
Add /v1/poems/comment/delete to allow User to delete Comment made by them or to their Poem

#28 #30 #32

### CODE REVIEW

File: controllers/poemController.js [hunk 1]
```diff
@@ -129,11 +130,11 @@ exports.visit = async function(params) {
   // Update viewCount for Poem and User
   await Promise.all([
     Poem.findByIdAndUpdate(params.poemId,
-        {$inc: {viewCount: 1}
-      }),
+        {$inc: {viewCount: 1},
+        }),
```
Comment (Kailang) [inline controllers/poemController.js]: In a new PR, start a newline after the open curly bracket.
```
    Poem.findByIdAndUpdate(params.poemId, {
      $inc: {viewCount: 1}
    }),
```

Comment (Kailang) [reply controllers/poemController.js]: Revert unrelated changes.

File: controllers/poemController.js [hunk 2]
```diff
@@ -164,3 +165,42 @@ exports.detail = async function(params) {
...
+  return apiSuccess();
+};
+
+exports.commentDelete = async function(params) {
+  const userId = tokenService.getUserId(params.token);
+  const comment = await Comment.findById(params.commentId);
+  if (!comment) {
+    return apiError(NOT_FOUND);
+  }
+  if ((userId !== comment.commentAuthorId) && (userId !== comment.poemAuthorId)) {
```
Comment (Kailang) [inline controllers/poemController.js]: Useless brackets.
`userId !== comment.commentAuthorId && userId !== comment.poemAuthorId`

File: controllers/poemController.js [hunk 3]
```diff
@@ -164,3 +165,42 @@ exports.detail = async function(params) {
...
+  await Comment.create({
+    poemAuthorId: poem.authorId,
+    commentAuthorId: userId,
+    poemId: params.poemId,
+    body: params.comment,
+    date: params.date,
+  });
+  Poem.findByIdAndUpdate(params.poemId,
+      {$inc: {commentCount: 1},
+      });
```
Comment (Kailang) [inline controllers/poemController.js]: Poem.findByIdAndUpdate(...).exec();
Also add comment about not awaiting since it does not affect the response.

Comment (HinataIzayoi) [reply controllers/poemController.js]: Done.

File: controllers/poemController.js (hunk 2 above)
Comment (HinataIzayoi) [reply controllers/poemController.js]: Done.

File: controllers/poemController.js (hunk 1 above)
Comment (HinataIzayoi) [reply controllers/poemController.js]: Done.

File: controllers/poemController.js [hunk 4]
```diff
@@ -159,8 +159,48 @@ exports.detail = async function(params) {
     return apiError(UNAUTHORIZED);
   }
   const userId = tokenService.getUserId(params.token);
-  if (poem.authorId === userId) {
+  if (poem.authorId == userId) {
```
Comment (Kailang) [inline controllers/poemController.js]: Use `===`.

File: controllers/poemController.js [hunk 5]
```diff
@@ -159,8 +159,48 @@ exports.detail = async function(params) {
...
+  });
+  // No await since commentCount is not used in the response
+  Poem.findByIdAndUpdate(params.poemId, {
+    $inc: {commentCount: 1},
+  }).exec();
+
+  return apiSuccess();
+};
+
+exports.commentDelete = async function(params) {
```
Comment (Kailang) [inline controllers/poemController.js]: Use `deleteComment`

File: controllers/poemController.js [hunk 6]
```diff
@@ -159,8 +159,48 @@ exports.detail = async function(params) {
...
   }
   const userId = tokenService.getUserId(params.token);
-  if (poem.authorId === userId) {
+  if (poem.authorId == userId) {
     return apiSuccess(poem);
   }
   return apiError(FORBIDDEN);
 };
+
+exports.comment = async function(params) {
```
Comment (Kailang) [inline controllers/poemController.js]: Use `createComment`

File: controllers/poemController.js (hunk 4 above)
Comment (HinataIzayoi) [reply controllers/poemController.js]: Does not seem to work with ===

File: controllers/poemController.js (hunk 5 above)
Comment (HinataIzayoi) [reply controllers/poemController.js]: Done.

File: controllers/poemController.js (hunk 6 above)
Comment (HinataIzayoi) [reply controllers/poemController.js]: Done.

**Pull request merged**